### PR TITLE
Support fields with collections of references in Kotlin

### PR DESCRIPTION
**Pull request by csilvestrini**
Diffbase #5025

### CODE REVIEW

File: src/tools/tests/goldens/generated-schemas.jvm.kt [hunk 1]
```diff
@@ -247,7 +247,7 @@ class Gold_Data(
                 ),
                 collections = emptyMap()
```
Comment (jblebrun) [inline src/tools/tests/goldens/generated-schemas.jvm.kt]: Same question as in #5025

**Comment by mmandlis** [comment]
qq, once both these prs landed, is it ok to go ahead and update the internal schemas?

File: src/tools/schema2kotlin.ts [hunk 2]
```diff
@@ -39,27 +39,35 @@ export interface KotlinTypeInfo {
...
   }
-  if (name === 'Reference') {
-    assert(refClassName, 'refClassName must be provided for References');
-    assert(refSchemaHash, 'refSchemaHash must be provided for References');
+  if (opts.name === 'Reference') {
+    assert(opts.refClassName, 'refClassName must be provided for References');
+    assert(opts.refSchemaHash, 'refSchemaHash must be provided for References');
+    if (!opts.isCollection) {
+      // Singleton Reference fields are nullable.
+      info.type += '?';
```
Comment (cromwellian) [inline src/tools/schema2kotlin.ts]: Did anyone ever discuss the alternate, a Reference that always returns null when you deref it? Or as a Collection with only 0 or 1 refs? I suppose in Kotlin with first class nullability support, boxed types are frowned upon (e.g. Optional/Maybe types), but it does allow you do write code that map()s over fields without having to sprinkle null-safe operators.

**Comment by csilvestrini** [comment]
> qq, once both these prs landed, is it ok to go ahead and update the internal schemas?

Yep, you should be good to go!

File: src/tools/schema2kotlin.ts (hunk 2 above)
Comment (csilvestrini) [reply src/tools/schema2kotlin.ts]: It hasn't come up so far. But Kotlin makes dealing with nulls so easy that I'm not sure it's worth it tbh.

**Pull request merged**